File: api/webhook.js
// Italian Verbs Telegram Bot - Vercel webhook style

// Store verbs in memory
const verbs = [
  { id: 1, italian: "andare", english: "to go" },
  { id: 2, italian: "avere", english: "to have" },
  { id: 500, italian: "promettere", english: "to promise" },
];

// Store user progress in memory
const userProgress = new Map();

// Main webhook handler
export default async function handler(req, res) {
  if (req.method !== "POST") {
    return res.status(200).send("Bot running");
  }

  const { message } = req.body;
  const chatId = message?.chat?.id;
  const text = message?.text?.trim();

  if (!chatId || !text) return res.status(200).end();

  // START command
  if (text === "/start" || text === "/restart") {
    userProgress.set(chatId, 0);
    await askQuestion(chatId);
    return res.status(200).end();
  }

  // ANSWER handling
  let currentIndex = userProgress.get(chatId) ?? 0;

  if (currentIndex < verbs.length) {
    const verb = verbs[currentIndex];
    const correctAnswer = verb.english.toLowerCase();
    const userAnswer = text.toLowerCase();

    if (userAnswer === correctAnswer) {
      await sendMessage(
        chatId,
        `✅ Correct! **${verb.italian}** means **${verb.english}**.`
      );
    } else {
      await sendMessage(
        chatId,
        `❌ Not quite. **${verb.italian}** means **${verb.english}**.`
      );
    }

    // Move to next question
    userProgress.set(chatId, currentIndex + 1);
    await askQuestion(chatId);
  } else {
    await sendMessage(
      chatId,
      "🎉 Congratulations! You've completed all 500 verbs!"
    );
    userProgress.set(chatId, 0);
  }

  res.status(200).end();
}

// Ask next question
async function askQuestion(chatId) {
  const index = userProgress.get(chatId) ?? 0;
  if (index < verbs.length) {
    const verb = verbs[index];
    await sendMessage(
      chatId,
      `(${verb.id}/${verbs.length}) What does **${verb.italian}** mean?`
    );
  }
}

// Send message to Telegram
async function sendMessage(chatId, text) {
  await fetch(
    `https://api.telegram.org/bot${process.env.BOT_TOKEN}/sendMessage`,
    {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        chat_id: chatId,
        text,
        parse_mode: "Markdown",
      }),
    }
  );
}

File: api/webhook.py
import os
import logging
from telegram import Update
from telegram.ext import Application, CommandHandler, MessageHandler, filters, ContextTypes

logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s', level=logging.INFO)
logger = logging.getLogger(__name__)

# Store token from environment or fallback (not recommended for production)
BOT_TOKEN = os.environ.get("BOT_TOKEN", "8279668033:AAGE9Mg9k20tQVXRHCzY1eVhVDlojvKuNic")

# Verb list
verbs = [
    {"id": 1, "italian": "andare", "english": "to go"},
    {"id": 2, "italian": "avere", "english": "to have"},
    {"id": 500, "italian": "promettere", "english": "to promise"}
]

# Store user progress
user_progress = {}

async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):
    chat_id = update.effective_chat.id
    user_progress[chat_id] = 0
    await ask_question(update, context)

async def ask_question(update: Update, context: ContextTypes.DEFAULT_TYPE):
    chat_id = update.effective_chat.id
    current_index = user_progress.get(chat_id, 0)
    if current_index < len(verbs):
        verb = verbs[current_index]
        await update.message.reply_text(
            f"({verb['id']}/{len(verbs)}) What does **{verb['italian']}** mean?"
        )
    else:
        await update.message.reply_text("🎉 Congratulations! You've completed all 500 verbs!")
        user_progress[chat_id] = 0

async def handle_answer(update: Update, context: ContextTypes.DEFAULT_TYPE):
    chat_id = update.effective_chat.id
    current_index = user_progress.get(chat_id, 0)
    if current_index < len(verbs):
        verb = verbs[current_index]
        user_answer = update.message.text.lower().strip()
        correct_answer = verb['english'].lower()

        if user_answer == correct_answer:
            response = f"✅ Correct! **{verb['italian']}** means **{verb['english']}**."
        else:
            response = f"❌ Not quite. **{verb['italian']}** means **{verb['english']}**."

        await update.message.reply_text(response)
        user_progress[chat_id] = current_index + 1
        await ask_question(update, context)

async def webhook(request):
    """Vercel webhook handler."""
    application = Application.builder().token(BOT_TOKEN).build()
    application.add_handler(CommandHandler("start", start))
    application.add_handler(CommandHandler("restart", start))
    application.add_handler(MessageHandler(filters.TEXT & ~filters.COMMAND, handle_answer))

    if request.method == "POST":
        data = await request.json()
        await application.update_queue.put(Update.de_json(data, application.bot))
        return {"statusCode": 200, "body": "ok"}
    else:
        return {"statusCode": 200, "body": "Bot is running"}

# Required for Vercel
handler = webhook
